src/biogeochem/ch4Mod.F90

```diff
@@ -1554,13 +1560,61 @@ subroutine readParams ( ncid )
   end subroutine readParams
 
   !-----------------------------------------------------------------------
-  subroutine ch4_init_balance_check(bounds, num_nolakec, filter_nolakec, num_lakec, filter_lakec, &
+  subroutine ch4_init_gridcell_balance_check(bounds, num_nolakec, &
+                filter_nolakec, num_lakec, filter_lakec, ch4_inst)
+    !
+    ! !DESCRIPTION:
+    ! Calculate beginning gridcell-level ch4 balance for mass conservation
+    ! check
+    !
+    ! This sets ch4_inst%totcolch4_bef_grc
+    !
+    ! Called before the weight updates done for dynamic landunits and the
+    ! associated filter updates
+    !
+    ! !USES:
+    use subgridAveMod, only: c2g
+    !
+    ! !ARGUMENTS:
+    type(bounds_type), intent(in)    :: bounds
+    integer          , intent(in)    :: num_nolakec        ! number of column non-lake points in column filter
+    integer          , intent(in)    :: filter_nolakec(:)  ! column filter for non-lake points
+    integer          , intent(in)    :: num_lakec          ! number of column lake points in column filter
+    integer          , intent(in)    :: filter_lakec(:)    ! column filter for lake points
+    type(ch4_type)   , intent(inout) :: ch4_inst
+    !
+    ! !LOCAL VARIABLES:
+
+    character(len=*), parameter       :: subname = 'ch4_init_gridcell_balance_check'
+    !-----------------------------------------------------------------------
+
+    ! Initialize to zero for columns outside the filters because will
+    ! average up to gridcell
+    ch4_inst%totcolch4_bef_col(bounds%begc:bounds%endc) = 0._r8
```

Comment:
Removing this works. To explain though, I preemptively inserted this and the other initialization because I was burned in #984 debugging for what ended up being partly undefined column variables entering the c2g subroutine.

Reply:
Makes sense. And truly, there isn't really a "right" answer here: the tradeoff between performance and being extra-safe is rarely a clear one.